src/main/java/id/shivelight/paper/unexpectedspawn/Spawn.java

```diff
@@ -37,52 +38,191 @@
 
     private final UnexpectedSpawn plugin;
     private final HashSet<Material> blacklistedMaterial = new HashSet<>();
+    private final HashSet<World> blackListedWorlds = new HashSet<>();
 
     Spawn(UnexpectedSpawn plugin) {
         this.plugin = plugin;
-        List<String> materialList = plugin.config.getConfig().getStringList("spawn-block-blacklist");
-        for (String name : materialList) {
-            Material material = Material.getMaterial(name);
-            if (material == null) {
-                Bukkit.getLogger().warning("Material " + name + " is not valid. See https://papermc.io/javadocs/paper/org/bukkit/Material.html");
+
+//        List<String> materialList = plugin.config.getConfig().getStringList("global.spawn-block-blacklist");
+//        for (String name : materialList) {
+//            Material material = Material.getMaterial(name);
+//            if (material == null) {
+//                Bukkit.getLogger().warning("Material " + name + " is not valid. See https://papermc.io/javadocs/paper/org/bukkit/Material.html");
+//                continue;
+//            }
+//            blacklistedMaterial.add(material);
+//        }
+
+        List<String> worldList = plugin.config.getConfig().getStringList("blacklisted-worlds");
+        for (String name: worldList) {
+            World world = Bukkit.getWorld(name);
+            if (world == null) {
+                logMessageConsole("Couldn't find world "+ name + ". Either it doesn't exist or is not valid.","warn");
                 continue;
             }
-            blacklistedMaterial.add(material);
+            blackListedWorlds.add(world);
+        }
+    }
+
+    private void logMessageConsole(String msg, String type) {
+        String logPrefix = "[UnexpectedSpawn] ";
+        if (type.equals("severe")) {
+            Bukkit.getLogger().severe(logPrefix + msg);
+        }
+        else if (type.equals("warn")) {
+            Bukkit.getLogger().warning(logPrefix + msg);
+        }
+        else {
+            Bukkit.getLogger().info(logPrefix + msg);
```

Comment:
`type` parameter can be an enum. Furthermore, it is a good idea to extract this method to a new class and reuse it.

Reply:
Will check it and try implementing it. Thank You for feedback.

Reply:
Even better, create methods for each level, e.g. Logger#warning, Logger#info, etc.

Reply:
So normal class or enum class?
Ok per log method.

Reply:
I was thinking both, but per log level methods is much better?

Reply:
So for enum one java file and for logger methods one?
You can dm me in discord also.

Reply:
It's done. +resolved